files/xml.py

```diff
@@ -0,0 +1,470 @@
+    # Check if the file exists
+    # No: abort
+    if xml_string:
+        infile = BytesIO(xml_string.encode('utf-8'))
+    elif os.path.isfile(xml_file):
+        infile = file(xml_file, 'r')
+    else:
+        module.fail_json(
+            msg="The target XML source does not exist: %s" %
+            xml_file)
+
+    # Try to parse in the target XML file
+    try:
+        parser = etree.XMLParser(remove_blank_text=pretty_print)
+        x = etree.parse(infile, parser)
+    except etree.XMLSyntaxError, e:
+        module.fail_json(
+            msg="Error while parsing file: %s" %
+            str(e))
+
+    if module.params['print_match']:
+        print_match(module, x, xpath, namespaces)
+
+    if module.params['count']:
+        count(module, x, xpath, namespaces)
+
+    if content == 'attribute':
+        get_element_attr(module, x, xpath, namespaces)
+    elif module.params['content'] == 'text':
+        get_element_text(module, x, xpath, namespaces)
+
+    # module.fail_json(msg="OK. Well, etree parsed the xml file...")
+
+    # module.exit_json(what_did={"foo": "bar"}, changed=True)
+
+    ##################################################################
+    # File exists:
+    # Ensure:
+    if ensure == 'absent':
+        # - absent: delete xpath target
+        delete_xpath_target(module, x, xpath, namespaces)
+        # Exit
+    # - present: carry on
+
+    ##################################################################
+    # children && value both set?: should have already aborted by now
+    ##################################################################
+
+    ##################################################################
+    # add_children && set_children both set?: should have already aborted by now
+    ##################################################################
+
+    ##################################################################
+    # set_children set?
+    # Yes: Set children of target
+    if module.params['set_children']:
+        set_target_children(module, x, xpath, namespaces, set_children)
+
+    ##################################################################
+    # add_children set?
+    # Yes: Add children to target
+    if module.params['add_children']:
+        add_target_children(module, x, xpath, namespaces, add_children)
+
+    # No?: Carry on
+
+    ##################################################################
+    # Is the xpath target an attribute selector?
+    # Yes: Set the attribute, exit
+    if module.params['value'] is not None:
+        set_target(module, x, xpath, namespaces, attribute, value)
+
+    ##################################################################
+    # Format the xml only?
+    if module.params['pretty_print']:
+        pretty(module, x)
+
+    abort(module, "don't know what to do")
+
+######################################################################
+from ansible.module_utils.basic import *
```

Comment:
This can now be at the top of the file and no longer needs to be a wildcard import for Ansible-2.1 or later.